[Cheng Deren] iP

DerenC opened the pull request:
# Wessy, your friendly neighbourhood personal assistant
What I cannot do:
* Take your *trash* out
* Eat your *vege*
* Do your *homework*

What I can do:
1. Help you keep track of the **task** you need to do
2. Help you keep track of the **event** you need to attend
3. Help you keep track of upcoming **deadlines**

I can even print `hello world`, just like you.
```
class Main {
    public static void main(String[] args) {
        System.out.println("Hello World");
    }
}
```

What you should do everyday:
- [ ] Greet me every morning.
- [x] Use me more and more :smirk:
- [ ] Say goodbye to me every night.

~~BEND DOWN TO ME, YOU LOWLY HUMAN~~

> "Wubba Lubba Dub Dub" ~ Rick Sanchez ([source](https://rickandmorty.fandom.com/wiki/Wubba_Lubba_dub-dub))

Commit:
Add Gradle support

DerenC committed:
Add *.class to .gitignore

DerenC committed:
Finish Level-1

DerenC committed:
Finish Level-2

DerenC committed:
Change the chatbot's name from Duke to Wessy

DerenC committed:
Improve the chatbot's opening message

DerenC committed:
Increase indentation to improve the appearance of the displayed message

DerenC committed:
Create 'Task' class

DerenC committed:
Finish Level-3

DerenC committed:
Finish Level-4

DerenC committed:
Clear up unused add functions

DerenC committed:
Finish Level-5

DerenC committed:
Modify EXPECTED.TXT & input.txt

DerenC committed:
Finish Level-6

DerenC committed:
Use enumerations instead of strings in Wessy (main driver) class

DerenC committed:
Complete checkstyle based on CS2030's standard

anchengyang commented on src/main/java/Wessy.java:
```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+    static String[] parse(String description, CmdType type) throws
```
Would it be better to create a Parser class instead?

anchengyang commented on src/main/java/Wessy.java:
```diff
@@ -0,0 +1,279 @@
+        return new String[] {};
+    }
+
+    static boolean checkCmd(String userInput, CmdType type) {
```
Maybe all these checks can be abstracted away from the main file? 🤔

anchengyang commented on src/main/java/Wessy.java:
```diff
@@ -0,0 +1,279 @@
+        return idx;
+    }
+
+    static void markOrUnmark(String userInput, boolean isMark) throws
```
Maybe these functions can also be abstracted away into a mark event class for example?

anchengyang reviewed (commented):
Looks good just a few OOP suggestions 😄

DerenC committed:
Finish Level-7

DerenC committed:
Finish Level-8

DerenC committed:
Merge branch 'branch-Level-7'

DerenC committed:
Merge branch 'branch-Level-8'

DerenC committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

seriouslia0 commented on src/main/java/Wessy.java:
```diff
@@ -89,6 +92,42 @@ static String[] parse(String description, CmdType type) throws
         return new String[] {};
     }
 
+    static LocalDateTime parseDateTime(String str) {
```
Might want to store this parser into a seperate class

seriouslia0 commented on src/main/java/Wessy.java:
```diff
@@ -194,22 +233,38 @@ static void printAdded(Task task) {
+            System.out.println("   |   " + str.substring(0, 64) + "   |");
+            int remainingLength = length - 64;
+            int leftover = remainingLength % 62;
+            int n = (int) Math.floor(remainingLength/62);
```
Might want to split this function into several to make it easier to edit in the future.

seriouslia0 reviewed (approved):
Overall LGTM

DerenC committed:
Initial setup of Gradle

wpx12011 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.LocalDateTime;
+
+public class Deadline extends Task {
```
Perhaps can put this class into a package (like package duke), as other developers could better understand the code base when all the classes have been grouped in packages

wpx12011 commented on src/main/java/Wessy.java:
```diff
@@ -0,0 +1,397 @@
+import Exceptions.CommandNotFoundException;
```
Great job on importing the classes explicitly, and ordered them in a logical way.

wpx12011 reviewed (commented):
Good job for following the coding standard closely! Perhaps can implement a Parser class to reduce the amount of contents in the Wessy class.

DerenC committed:
Finish A-MoreOOP

DerenC committed:
Refactor the code by applying more OOP principle

Refactor the whole code base by extracting out the closely related code
as the following components, with their respective function
* Parser -- Process user input.
* Storage -- Retrieve from existing storage and save as a new file after
  every update.
* Ui -- Handle I/O.
* UserInputChecker -- Check and throw exceptions while checking for
  correct formatting in the user input.

Rethink and recategorise the exceptions by merging those that seem
closely related. Add TimeSpecifierException on tp of
UnspecifiedTimeException to differentiate between the 2 exceptions.

Create more packages. Rearrange the code by placing the relevant classes into their respective
packages.

DerenC committed:
A-Packages

DerenC committed:
Add JUnit test to test getCmd(), parseDateTime(), removeSpacePadding() methods in Parser

DerenC committed:
Create jar file for version release

DerenC committed:
Add Java documentation for some classes.
These classes are:
	Parser
	UserInputChecker
	CmdType
	Task
	ToDo
	Deadline
	Event
	WessyException
	CommandNotFoundException
	MissingSpacingException
	TimeSpecifierException
	UnspecifiedTimeException
	EmptyListException
	InvalidIntegerException
	NotAnIntegerException
	NotPositiveIntegerException
	MissingInputException
	TooManyInputException

DerenC committed:
Finish A-CodingStandard

DerenC committed:
Finish Level-9

DerenC committed:
Merge branch 'branch-Level-9'

DerenC committed:
Merge branch 'branch-A-JavaDoc'

DerenC committed:
Merge branch 'branch-A-CodingStandard'

DerenC committed:
Add a basic GUI to Wessy
Use JavaFX technology to implement the GUI.

DerenC committed:
Rename methods in Ui from prints to gets

DerenC committed:
Merge commit 'cd2818dd855ef73b919892fe0703f1bef09f8db5'

DerenC committed:
Delete unnecessary java files outside the main.java.wessy directory

DerenC committed:
Use Assertions

Use assert feature (not JUnit assertions) to document importaant
assumptions that should hold at various points in the code.

DerenC committed:
Improve code quality

DerenC committed:
Use Streams

Use the Streams feature of Java in my code, if applicable.

DerenC committed:
Merge pull request #4 from DerenC/branch-A-Streams

Use Streams

DerenC committed:
Merge branch 'master' into branch-A-Assertions

DerenC committed:
Merge pull request #2 from DerenC/branch-A-Assertions

Use Assertions

DerenC committed:
Merge branch 'master' into branch-A-CodeQuality

DerenC committed:
Merge pull request #3 from DerenC/branch-A-CodeQuality

Improve code quality

DerenC committed:
Delete commented-out code used for Ui in Wessy.java

DerenC committed:
Declutter by deleting commented-out codes

DerenC committed:
Comment out System.in as standardInput

DerenC committed:
Implemented 2 extensions from category B

Implement the following 2 features:
* B-DoAfterTasks
* B-FixedDurationTasks

DerenC committed:
Add a User Guide

DerenC committed:
Minor correction on phrasing of descrption of Feature #3

DerenC committed:
Second minor correction on phrasing of descrption of Feature #3
